text/0000-Project-Goals-2025h2.md

```diff
@@ -0,0 +1,754 @@
+| [Stabilize public/private dependencies](https://rust-lang.github.io/rust-project-goals/2025h2/pub-priv.html)                                                       | ![Help Wanted][] | [cargo] ([Ed Page][]), [compiler]                                                          |
+| [Expand the Rust Reference to specify more aspects of the Rust language](https://rust-lang.github.io/rust-project-goals/2025h2/reference-expansion.html)           | [Josh Triplett][]    | [lang-docs] ([Josh Triplett][]), [spec] ([Josh Triplett][])                                   |
+| [reflection and comptime](https://rust-lang.github.io/rust-project-goals/2025h2/reflection-and-comptime.html)                                                      | [Oliver Scherer][]         | [compiler] ([Oliver Scherer][]), [lang] ([Scott McMurray][]), [libs] ([Josh Triplett][])                     |
+| [Relink don't Rebuild](https://rust-lang.github.io/rust-project-goals/2025h2/relink-dont-rebuild.html)                                                             | [Jane Lusby][]           | [cargo], [compiler]                                                                   |
+| [Rust Vision Document](https://rust-lang.github.io/rust-project-goals/2025h2/rust-vision-doc.html)                                                                 | [Niko Matsakis][]    | [leadership-council]                                                                  |
+| [rustc-perf improvements](https://rust-lang.github.io/rust-project-goals/2025h2/rustc-perf-improvements.html)                                                      | [James][]    | [compiler], [infra]                                                                   |
+| [Stabilize rustdoc `doc_cfg` feature](https://rust-lang.github.io/rust-project-goals/2025h2/rustdoc-doc-cfg.html)                                                  | [Guillaume Gomez][]  | [rustdoc] ([Tugce Ciftci][])                                                                  |
+| [Add a team charter for rustdoc team](https://rust-lang.github.io/rust-project-goals/2025h2/rustdoc-team-charter.html)                                             | [Guillaume Gomez][]  | [rustdoc] ([Tugce Ciftci][])                                                                  |
+| [SVE and SME on AArch64](https://rust-lang.github.io/rust-project-goals/2025h2/scalable-vectors.html)                                                              | [David Wood][]       | [compiler] ([David Wood][]), [lang] ([Niko Matsakis][]), [libs] ([Amanieu d'Antras][]), [types]           |
+| [Rust Stabilization of MemorySanitizer and ThreadSanitizer Support](https://rust-lang.github.io/rust-project-goals/2025h2/stabilization-of-sanitizer-support.html) | [Jakob Koschel][]       | [bootstrap], [compiler], [infra], [project-exploit-mitigations]                       |
```

Comment:
In [#t-infra>2025H2 Goal Review](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/2025H2.20Goal.20Review/with/532155078) the question was raised if this actually requires infrastructure support or isn't just another bootstrap component. Has that question been answered yet?

Reply:
I think that the only infra thing about this is that we would ship a new rustup component, and thus have more traffic, in theory. Shouldn't be an issue, I suppose.

Reply:
It's not just a single component, it's a component for each sanitizer and each tier 1 target, right?

And `-Zbuid-std` is also a goal so in the not-too-far-future it might not even be necessary any more to ship these prebuilt standard libraries.

Reply:
I think the latest consensus on Zulip was to ship all sanitizers in a single component. Ofc the component will be specialized per target, I don't see that as being "multiple components".

Reply:
Okay I see.

For CI it still means a lot of extra std builds either way.

Reply:
Yeah I think it's fine. I think it was mostly just a question of "does this *even* need any t-infra participation", it wasn't a specific concern about infra stuff.

Reply:
Yeah, I think that works.